handlers/kannel/handler.go

```diff
@@ -200,10 +201,20 @@ func (h *handler) Send(ctx context.Context, msg courier.MsgOut, clog *courier.Ch
 	}
 
 	var resp *http.Response
+	var respBody []byte
 	if verifySSL {
-		resp, _, err = h.RequestHTTP(req, clog)
+		resp, respBody, err = h.RequestHTTP(req, clog)
 	} else {
-		resp, _, err = h.RequestHTTPInsecure(req, clog)
+		resp, respBody, err = h.RequestHTTPInsecure(req, clog)
+	}
+
+	if strings.Contains(string(respBody), "Queued") {
+		rc := h.Backend().RedisPool().Get()
+		defer rc.Close()
+		rateLimitKey := fmt.Sprintf("rate_limit:%s", msg.Channel().UUID())
```

Comment:
It doesn't feel ideal having handlers talk directly to Redis to rate limit themselves and I see we only currently do this in the old WhatsApp onprem handler. And there's `rate_limit_bulk`.. don't we need to rate limit on both queues?

Maybe we need to have a think about making a cleaner way for channels to rate limit themselves?

Reply:
Okay, we can find a new way tor that.

I do not think we need to pause the bulk queue in this case but I will double check on that again

Reply:
Added pausing the bulk queue as well